pkg/osquery/table/user_avatar_darwin.go

```diff
@@ -0,0 +1,93 @@
+package table
+
+/*
+#cgo darwin CFLAGS: -DDARWIN -x objective-c
+#cgo darwin LDFLAGS: -framework Cocoa -framework Collaboration
+#import <Collaboration/Collaboration.h>
+#include <CoreFoundation/CoreFoundation.h>
+void Image(CFDataRef *imageDataRef, char* user) {
+NSString *userName = [NSString stringWithFormat:@"%s", user];
+CBIdentity *identity = [CBIdentity identityWithName:userName authority:[CBIdentityAuthority defaultIdentityAuthority]];
+NSImage *userImage = [identity image];
+NSData *imageData = [userImage TIFFRepresentation];
+*imageDataRef = (CFDataRef)imageData;
+}
+*/
+import "C"
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"image/png"
+	"unsafe"
+
+	osquery "github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/nfnt/resize"
+	"github.com/pkg/errors"
+	"golang.org/x/image/tiff"
+)
+
+func UserAvatar(client *osquery.ExtensionManagerClient) *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("username"),
+		table.TextColumn("thumbnail"),
+	}
+	t := &userAvatarTable{client: client}
+	return table.NewPlugin("kolide_user_avatar", columns, t.generateAvatar)
+}
+
+type userAvatarTable struct {
+	client      *osquery.ExtensionManagerClient
+	primaryUser string
+}
+
+func (t *userAvatarTable) generateAvatar(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+
+	if t.primaryUser == "" {
+		username, err := queryPrimaryUser(t.client)
+		if err != nil {
+			return nil, errors.Wrap(err, "query primary user for airdrop table")
+		}
+		t.primaryUser = username
+	}
+
+	// use the username from the query context if provide, otherwise default to primary user
+	var username string
+	q, ok := queryContext.Constraints["username"]
+	if ok && len(q.Constraints) != 0 {
+		username = q.Constraints[0].Expression
+	} else {
+		username = t.primaryUser
+	}
+
+	var data C.CFDataRef = 0
+	C.Image(&data, C.CString(username))
+	defer C.CFRelease(C.CFTypeRef(data))
+	goBytes := C.GoBytes(unsafe.Pointer(C.CFDataGetBytePtr(data)), C.int(C.CFDataGetLength(data)))
```

Comment:
Is there any CF function that could provide the bytes in some sort of type that would allow us to avoid the `unsafe.Pointer`?